### PR TITLE
Minimal s-expression ast generation

### PR DESCRIPTION
**Pull request by Danielhu229**

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+fn empty() {
+    let empty_tokens: Vec<Token> = Vec::new();
+    let mut parser = Parser::new(empty_tokens.into_iter());
+    let ast = parser.parse().unwrap();
+    assert_eq!(ast, None);
+}
+
+#[test]
+fn number() {
+    let empty_tokens: Vec<Token> = vec![Token::Number(1)];
```
Comment (miloyip) [inline src/parser.rs:90]: It is not `empty_tokens`. They may be named simply `tokens` for all tests.

Comment (Danielhu229) [reply src/parser.rs:90]: Yes, will update later.

Comment (Danielhu229) [reply src/parser.rs]: Updated

**Comment by miloyip** [comment]
Tests should also cover error routes, such as `SyntaxError`

**Comment by Danielhu229** [comment]
Updated.

**Pull request merged**